boards/bitcraze/crazyflie/syslink/syslink_main.cpp

```diff
@@ -404,9 +404,10 @@ Syslink::handle_message(syslink_message_t *msg)
 		memcpy(&vbat, &msg->data[1], sizeof(float));
 		//memcpy(&iset, &msg->data[5], sizeof(float));
 
-		_battery.updateBatteryStatus(t, vbat, -1, true,
-					     battery_status_s::BATTERY_SOURCE_POWER_MODULE, 0, 0);
-
+		_battery.setConnected(true);
+		_battery.updateVoltage(vbat);
+		_battery.updateBatteryStatus(t);
+		_battery.publishBatteryStatus(_battery.getBatteryStatus());
```

Comment:
I'd add a convenience method, to reduce this a bit (same pattern in many places):
```suggestion
		_battery.updateAndPublishBatteryStatus(t);
```

Reply:
Good idea, I added it directly in c12d272